### PR TITLE
Perceptual

### PR DESCRIPTION
**Pull request by 51N84D**
Adding patchgan, perceptual loss, and discriminator feature matching loss functionality

### CODE REVIEW

File: scripts/networks.py [hunk 1]
```diff
@@ -219,6 +221,189 @@ def calc_gen_loss_sr(self, input_fake):
         return loss
 
 
+class MultiscaleDiscriminator(nn.Module):
```
Comment (vict0rsch) [inline scripts/networks.py]: can you add the source? a comment with a link to the original code

File: scripts/networks.py [hunk 2]
```diff
@@ -528,61 +713,77 @@ def forward(self, x):
...
-        self.conv4_3 = nn.Conv2d(512, 512, kernel_size=3, stride=1, padding=1)
-
-        self.conv5_1 = nn.Conv2d(512, 512, kernel_size=3, stride=1, padding=1)
-        self.conv5_2 = nn.Conv2d(512, 512, kernel_size=3, stride=1, padding=1)
-        self.conv5_3 = nn.Conv2d(512, 512, kernel_size=3, stride=1, padding=1)
+from torchvision import models
+
+
+class Vgg19(torch.nn.Module):
+    def __init__(self, requires_grad=False):
```
Comment (vict0rsch) [inline scripts/networks.py]: same: source?

File: scripts/trainer.py [hunk 3]
```diff
@@ -435,6 +468,8 @@ def gen_update(
             if hyperparameters["vgg_w"] > 0:
                 comet_exp.log_metric("loss_gen_vgg_a", self.loss_gen_vgg_a.cpu().detach())
                 comet_exp.log_metric("loss_gen_vgg_b", self.loss_gen_vgg_b.cpu().detach())
+                # comet_exp.log_metric("loss_destruct_vgg_a", self.loss_destruct_vgg_a.cpu().detach())
```
Comment (vict0rsch) [inline scripts/trainer.py]: if this is not needed, maybe remove comment altogether?

File: scripts/networks.py (hunk 1 above)
Comment (51N84D) [reply scripts/networks.py]: Done

File: scripts/networks.py (hunk 2 above)
Comment (51N84D) [reply scripts/networks.py]: Also done

File: scripts/trainer.py (hunk 3 above)
Comment (51N84D) [reply scripts/trainer.py]: Removed
